### PR TITLE
Introducing Rubocop

### PR DESCRIPTION
**Pull request by abinashmeher999**

### CODE REVIEW

**Comment by abinashmeher999** [comment]
@v0dro I referred to the discussion here: https://github.com/SciRuby/nmatrix/issues/521 to set this up. Every violation passes for now. Could you please review?

**Comment by v0dro** [comment]
Yeah works. You can add rubocop to the Travis build so that it fails in case of violations and it can be easily seen on GitHub.

**Comment by isuruf** [comment]
@abinashmeher999, can you enable edits from maintainers?

**Comment by abinashmeher999** [comment]
Done.

**Comment by isuruf** [comment]
This is now ready for review

**Comment by certik** [comment]
@abinashmeher999, @zverok would you mind reviewing this please?

It's +1 from me.

**Comment by zverok** [comment]
Looking at it.

File: lib/symengine.rb [hunk 1]
```diff
@@ -25,45 +25,53 @@ def symbols ary_or_string, *params
 
       # Make an Array of SymEngine::Symbols from the parameters we received,
       # now that they're normalized.
-      ary_or_string.map do |symbol_or_string| 
+      ary_or_string.map do |symbol_or_string|
         SymEngine::Symbol.new(symbol_or_string)
       end
```
Comment (zverok) [inline lib/symengine.rb]: `ary_or_string.map(&SymEngine::Symbol.method(:new))`

File: spec/evalf_spec.rb [hunk 2]
```diff
@@ -1,27 +1,27 @@
 describe 'SymEngine::evalf' do
-    context 'RealDouble values' do
-        subject { SymEngine::evalf(SymEngine::sin(SymEngine(2)), 53, true) }
-        it { is_expected.to be_a SymEngine::RealDouble }
-        it { is_expected.to be_within(1e-15).of(0.909297426825682) }
+  context 'RealDouble values' do
+    subject { SymEngine.evalf(SymEngine.sin(SymEngine(2)), 53, true) }
+    it { is_expected.to be_a SymEngine::RealDouble }
+    it { is_expected.to be_within(1e-15).of(0.909297426825682) }
+  end
```
Comment (zverok) [inline spec/evalf_spec.rb]: Blank line here?..

File: spec/evalf_spec.rb [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
-        its(:imaginary) { is_expected.to be_within(1e-15).of(0.841470984807897) }
-    end
-    
-    if SymEngine::HAVE_MPC
-        context 'ComplexMPC values' do
-            subject { SymEngine::evalf(SymEngine::sin(1) * SymEngine::I, 100, false) }
-            it { is_expected.to  be_a SymEngine::ComplexMPC }
-            its(:imaginary) { is_expected.to be_within(SymEngine::RealMPFR.new("1e-32", 100)).of(SymEngine::RealMPFR.new("0.84147098480789650665250232163005", 100)) }
-        end
+  end
```
Comment (zverok) [inline spec/evalf_spec.rb]: ...and here

File: spec/functions_spec.rb [hunk 4]
```diff
@@ -29,17 +29,18 @@
...
+        let(:class_name) do
           f = fun.to_s
           if f.start_with?('a')
-            f.sub(/^../) {|ls| ls.upcase }
+            f.sub(/^../, &:upcase)
           else
             f.capitalize
           end
-        }
+        end
```
Comment (zverok) [inline spec/functions_spec.rb]: To be honest, for me the rule "`do..end` for multiline blocks" could be disabled at least for `spec/` folder.

File: spec/symbol_spec.rb [hunk 5]
```diff
@@ -36,13 +36,13 @@
...
   context 'with an array of Ruby Strings as an argument' do
-    let(:args) { [%w[x y]] } # it would be one Array argument on *args
+    let(:args) { [%w(x y)] } # it would be one Array argument on *args
 
     it_behaves_like 'two symbols'
   end
 
   context 'with a splatted argument' do
-    let(:args) { %w[x y] } # it would be several separate strings on *args
+    let(:args) { %w(x y) } # it would be several separate strings on *args
```
Comment (zverok) [inline spec/symbol_spec.rb]: Again, matter of personal style, but it is easy to redefine in `.rubocop.yml` what is appropriate limitation characters for `%w` arrays. And for me, `%w[]` is much more readable.

Comment (abinashmeher999) [inline spec/symbol_spec.rb]: +1

File: spec/functions_spec.rb (hunk 4 above)
Comment (abinashmeher999) [inline spec/functions_spec.rb]: Either is okay with me.

**Comment by abinashmeher999** [comment]
Thanks @isuruf!

**Comment by abinashmeher999** [comment]
Made the suggested changes.

**Comment by isuruf** [comment]
@abinashmeher999, thanks for the work and @zverok, @v0dro, @certik for the reviews

**Pull request merged**